modules/hostap/src/supp_main.c

```diff
@@ -307,7 +303,8 @@ static void iface_event_handler(struct net_mgmt_event_callback *cb,
 {
 	const char *ifname = iface->if_dev->dev->name;
 
-	if (strncmp(ifname, IFACE_MATCHING_PREFIX, sizeof(IFACE_MATCHING_PREFIX) - 1) != 0) {
+	if (strncmp(ifname, CONFIG_NRF700X_WLAN_IF_PREFIX,
+		    sizeof(CONFIG_NRF700X_WLAN_IF_PREFIX)) != 0) {
```

Comment:
This is big no-no, you are now introducing NRF specific checks into generic code. Beside zephyr upstream is already using interface name and not the device name to match the needed interface. It would be better if the downstream code would use the interface name too. If one knows network interface name, it is always possible to figure out what is the device, but it is not always possible to do the reverse (if VLAN is enabled).

Reply:
Moved the prefix definition to WPA supplicant module.

Reply:
Eventually the code in supp_main.c will be moved to upstream so we cannot have here NRF specific Kconfig symbols. So this cannot be done like this. Also wifi driver specific changes need to be separated to different commit than the hostap related ones.

Reply:
Done. Instead of NRF specific config name, using a WPA specific config. Separated the commits as well.